### ISSUE TITLE
Traumatic Wounds, Triage, and Treatment

### ISSUE THREAD

**Issue by PerfectTangent** [unitystation/unitystation#6272]
### Description:

**This is a health-related bounty.** This bounty calls for the addition of all the basic traumas, negative effects for having the traumas, and treatment methods to triage and cure said traumas.

Traumatic wounds have a chance to occur when a large amount of damage is accumulated on a single body part, or from particularly damaging attacks. These traumas require special treatment through the application of specific consumable items like sutures, gauze, regenerative meshes, surgical tape, bone gel, etc.

The 4 different surface traumas that can be caused by attacks and the 3 tiers of severity that can be caused are as follows:

Slash - the largest threat from Slashes should be bleeding. Tier 1 and 2 slashes will stop bleeding on their own after a time. Tier 3 slashes will not. Infection is a low chance.
- T1 Rough Abrasion
- T2 Open Laceration
- T3 Weeping Avulsion

Pierce - the largest threat from Punctures is bleeding.  Tier 2 and 3 punctures will not stop bleeding, but they don't bleed as profusely as Slash wounds. Infection is a low chance unless the pierce wound is in the chest, in which case it's medium.
- T1 Minor Breakage
- T2 Open Puncture
- T3 Ruptured Cavity

Blunt - the largest threat from blunt trauma is debilitation, or continuous organ damage if it is a fracture wound in the chest.
- T1 Joint Dislocation
- T2 Hairline Fracture
- T3 Compound Fracture

Burns - the largest threat from Burns should be infections.
- T1 Second Degree Burns
- T2 Third Degree Burns
- T3 Fourth Degree Burns (aka, Catastrophic Burns)

## What is needed to fulfill the bounty:

- [x] Add all 3 tiers for each of the 4 different kinds of traumatic wounds - Blunt, Pierce, Slash, and Burn.
- [x] Add a field to set what type of wounds a weapon deals. Weapons should be able to be configured to deal multiple kinds of traumatic wounds.
- [ ] Add debuffs and symptoms for traumatic wounds.
- [ ] Add the tools and items for performing triage and first aid on traumatic wounds (gauze, sutures, ointment, regen meshes).
- [ ] Add simple wound infections for leaving wounds (especially burns) untreated for too long.
- [x] Add sfx for when characters get dismembered or receive traumas.
- [x] Add a generalized way to trigger sfx based on what species a character is (lizardperson scream vs human scream, etc.)
- [ ] Add surgery for bones repairs, surgery for closing puncture wounds, and flesh debridement for treating severe infections.
- [ ] Update the UV pen light to be used for preventing infection or treating light infections.
- [x] Update mob and health analyzer examination text to reflect the traumas a character is experiencing.

## Notes

It would be very nice if there was a field for weapons that allows them to have a flexible threshold for having a chance to deal particular kinds of wounds (perhaps a multiplier?). This allows certain items to be particularly capable at causing traumas, which can be explored as a secondary statistic to think about when considering which weapons to use, so it's less of a _more damage beats less damage_ situation.

Bleeding should affect individual limbs and be treated individually rather than being a status effect on the whole body. For guns, the type of wounds they cause should be set in projectiles prefabs that they fire.

Information on the surgeries called for in this issue are described here: https://tgstation13.org/wiki/Surgery#Repair_Bone_Fracture_.28hairline.29

Information on the wounds called for in this issue are described here:
https://tgstation13.org/wiki/Guide_to_Wounds

**Comment by Cboyjet4** [comment]
no


we aren't based on modern tgmed, we are doing our own thing.
these will be a major annoyance, since we aren't having stasis beds, or surgery based revival.
wounds are unnecessary for this meds system.

**Comment by Bod9001** [comment]
humm,  one of the things I kept in mind with my work on health is that
you can patch yourself up to a reasonable degree mid fight, without any intervention from other people,
 however if you continue with the fight there will be organ damage that will accumulate over time meaning that you'll need surgical intervention to fully recover, this applies to mid-fight healing as well to get a perfect heel you would need surgical intervention on organ repair, I feel like moving the system down to the organ level would be a good idea, whilst not having it affect overall health

**Comment by PerfectTangent** [comment]
> humm, one of the things I kept in mind with my work on health is that you can patch yourself up to a reasonable degree mid fight, without any intervention from other people,

I agree with the general sentiment that negatives for having wounds should not directly interfere much with the actual capabilities of the player, with the exception of broken bones. I don't think traumas should affect overall health, either, unless they are bad wounds that are left untreated for too long. However, organ damage alone does not accurately simulate the kinds of wounds a body can take. 

> however if you continue with the fight there will be organ damage that will accumulate over time meaning that you'll need surgical intervention to fully recover, this applies to mid-fight healing as well to get a perfect heel you would need surgical intervention on organ repair, I feel like moving the system down to the organ level would be a good idea, whilst not having it affect overall health

The concept of this system is pretty much the same as that. It encourages people to perform triage and treatment, but doesn't really force it to be done at any particular time unless the character is close to death. The only difference is that it doesn't exclusively lean on organ damage. It uses other body functions to provide the same motivation. 

The reason a system like this is desireable is it increases the breadth of gameplay for medical officers in treatment. The actual debuffs from the problems are quite easily solved; every wound listed can be fixed either permanently or temporarily with gauze, with the exception of burn wounds which take ointment.

The point of the system is not to cause death spirals, where the longer it goes untreated, the odds of successfully surviving diminish exponentially. depending on where the threshold of damage needed to cause a wound is set, players likely won't even have to think about this system for most of the time they are playing. The point is to NOT make survival _harder,_ The point is to facilitate more interesting medical gameplay, one that rewards observant and prepared players.

**Comment by Bod9001** [comment]
for stuff like bleeding I feel that they should be a temporary battlefield fix, so you wrap it in a bandage and it no longer bleeds may be but you lose like 5 hit points maybe?

**Comment by PerfectTangent** [comment]
That is the solution in this system, Bod. Gauze fixes all issues with bleeding. In fact, in /tg/, bleeding treats itself in people even without any intervention unless it's a Tier 3 wound.

**Comment by Bod9001** [comment]
ok cool

**Comment by Bod9001** [comment]
note: You'll have to make a threshold for when puncture wounds start transferring to sub- organs, E.g  if your chest is horribly mangled then you're more likely to get organ damage if it gets punched again

**Comment by MaxIsJoe** [comment]
3 Of these have been fully implemented; The three damage tiers for Burn, Pierce and Slash damages and the ability to configure weapons to deal any of these damages.

Advanced Health Scanners can give a detailed report now by alt clicking on a targeted body part to see if a wound exists, if an organ is suffering from internal bleeding, what tier of damage the limb or organ has, how big wounds are and if that wound is causing the player to bleed.

**Comment by corp-0** [comment]
how is this going?

**Comment by MaxIsJoe** [comment]
Apparently this was edited to include blunt force trauma damage so I'll need to add that as well.
It's going great actually, but like I said in my last PR I have exams for a while so I'll be away for a week or two.

I'm adding bone fracturing right now and looking into how people are adding two torsos to the same body. I should also add in internal damage fixes because right now only the liver is the one that gets fixed (I think?)

**Comment by PerfectTangent** [comment]
> Apparently this was edited to include blunt force trauma damage so I'll need to add that as well.

It included blunt trauma from the beginning, Max. The concept for this system is basically pulled from /tg/ without changes.

**Comment by PerfectTangent** [comment]
If the outlined requirements to fulfill a bounty are ever changed or added to while someone is working on it, we would disclose it.

**Comment by MaxIsJoe** [comment]
I don't remember seeing blunt tho? Ah whatever, I needed a way to damage bones either way.

**Comment by pyrosshade** [comment]
Or rather, the singular bone. If you change the body so each part has a bone, make sure to change how much blood each bone makes accordingly.

**Comment by MaxIsJoe** [comment]
> Or rather, the singular bone. If you change the body so each part has a bone, make sure to change how much blood each bone makes accordingly.

Bones can bleed?

**Comment by pyrosshade** [comment]
They do not bleed, but rather are what generate the blood cells.

**Comment by MaxIsJoe** [comment]
>  Update mob and health analyzer examination text to reflect the traumas a character is experiencing.

Idk why this hasn't been updated when it's been added already.

**Comment by PerfectTangent** [comment]
you might find that would happen less often if you communicated directly with me more, Max.

**Comment by MaxIsJoe** [comment]
> you might find that would happen less often if you communicated directly with me more, Max.

![image](https://user-images.githubusercontent.com/34368774/132922565-457e7375-b097-4185-98cc-6ab76fa0c41f.png)
but I said this was added here before

**Comment by PerfectTangent** [comment]
again, you might find more success if you communicated **directly** with me.

**Comment by PerfectTangent** [comment]
also, does it work with regular examination? People should be able to visually identify wounds via examination.

**Comment by MaxIsJoe** [comment]
> also, does it work with regular examination? People should be able to visually identify wounds via examination.

The bounty said health analyzers only but I think this is an easy thing to add

**Comment by PerfectTangent** [comment]
it refers to mob examination.

> Update _mob_ and health analyzer examination text to reflect the traumas a character is experiencing.

**Comment by MaxIsJoe** [comment]
health analyzer examination should be "health analyzer **and** examination text" then

**Comment by PerfectTangent** [comment]
bounty partially paid out for work done so far.

**Comment by PerfectTangent** [comment]
after discussing with @MaxIsJoe, requirements were tweaked and the bounty was bumped slightly to account for it.

**Comment by PerfectTangent** [comment]
This bounty is being suspended temporarily so that we can ensure we are getting what we want out of the system.

**Comment by PerfectTangent** [comment]
Unassigning Max for the time being while they work on another bounty.

**Comment by corp-0** [comment]
The bounty status for this issue is suspended until new notice
